File: pr22.rb
def euler
  names = File.read("./p022_names.txt").gsub("\"", '').split(',').sort
  numbers = names.each_with_index.map do |name, i|
    (i + 1 ) * name.chars.map { |c| c.ord - 64 }.reduce(&:+)
  end
end

euler

File: pr22/Pr22.cs
using System;
using System.IO;

namespace Pr22
{
    public class Euler
    {
        public static void Main(string[] args)
        {
            string raw_input = File.ReadAllLines("./p022_names.txt")[0];
            string[] names = raw_input.Replace("\"","").Split(',');
            Array.Sort(names);
            
            int i = 0;
            int name_score;
            int total_score = 0;
            foreach(string name in names)
            {
                name_score = 0;

                foreach(char c in name)
                {
                    name_score += (int) c - 64;
                }

                total_score += name_score * ++i; 
            }

            Console.WriteLine(total_score);
        }
    }
}